### ISSUE TITLE
Move API documentation from Warehouse developer docs to PyPI user docs

### ISSUE THREAD

**Issue by di** [pypi/warehouse#16541]
Documentation for our APIs and datasets currently lives in our developer-facing docs: https://warehouse.pypa.io/api-reference/index.html

However, these are really end-user docs, and should live at https://docs.pypi.org/ instead.

**Comment by woodruffw** [comment]
Adding a roadmap:

- [ ] Index
    - [ ] Feeds
    - [ ] JSON API
    - [ ] Legacy API
    - [ ] Stats API
    - [ ] XML-RPC (N/A?)
    - [ ] Integration guide (N/A?)
    - [ ] BigQuery Datasets
    - [ ] Integrity API (i.e. PEP 740, not currently documented)

Once each of these is moved/added, it might make sense to add redirects from the old dev-docs URLs to the new user-docs ones.

**Comment by di** [comment]
> XML-RPC (N/A?)

I think it would be wise to not bother porting these docs.

> Integration guide (N/A?)

I think this can probably get included in the top-level page at https://docs.pypi.org/api/

**Comment by woodruffw** [comment]
All planned migrations are complete, so closing!

**Comment by di** [comment]
Should we add redirects from the old URLs?

**Comment by woodruffw** [comment]
> Should we add redirects from the old URLs?

Yep! I was trying to figure out a good way to do that in Sphinx but couldn't find one (there's a lot of ways to redirect to a _different_ Sphinx doc, but not an arbitrary URL). I'll do some more looking here.

**Comment by woodruffw** [comment]
https://documatt.com/sphinx-reredirects/usage.html might do the trick; I'll play around with that.

**Comment by woodruffw** [comment]
https://github.com/pypi/warehouse/pull/17213 has the redirects!
